src/status_im/components/toolbar_new/view.cljs

```diff
@@ -0,0 +1,107 @@
+                                                text-input
+                                                image
+                                                touchable-highlight]]
+            [status-im.components.sync-state.gradient :refer [sync-state-gradient-view]]
+            [status-im.components.styles :refer [icon-default
+                                                 icon-search]]
+            [status-im.components.context-menu :refer [context-menu]]
+            [status-im.components.toolbar-new.actions :as act]
+            [status-im.components.toolbar-new.styles :as st]
+            [status-im.accessibility-ids :as id]
+            [status-im.utils.platform :refer [platform-specific]]))
+
+(defn toolbar [{title                :title
+                nav-action           :nav-action
+                hide-nav?            :hide-nav?
+                actions              :actions
+                custom-action        :custom-action
+                background-color     :background-color
+                custom-content       :custom-content
+                style                :style}]
+  (let [style (merge (st/toolbar-wrapper background-color) style)]
+    [view {:style style}
+     [view st/toolbar
+      [view (st/toolbar-nav-actions-container actions)
+       (when-not hide-nav?
+         (if nav-action
+           [touchable-highlight {:on-press (:handler nav-action)}
+            [view
+             [image (:image nav-action)]]]
+           [touchable-highlight {:on-press            #(dispatch [:navigate-back])
+                                 :accessibility-label id/toolbar-back-button}
+            [view
+             [image {:source {:uri :icon_back_dark}
+                     :style  icon-default}]]]))]
+      (or custom-content
+          [view {:style st/toolbar-title-container}
+           [text {:style st/toolbar-title-text
+                  :font  :toolbar-title}
+            title]])
+      [view (st/toolbar-actions-container (count actions) custom-action)
+       (if actions
+         (for [{action-image   :image
+                action-options :options
+                action-handler :handler} actions]
+           (with-meta
+             (cond (= action-image :blank)
+                   [view st/toolbar-action]
+                   action-options
+                   [context-menu
+                    [view st/toolbar-action
+                     [image action-image]]
+                    action-options]
+                   :else
+                   [touchable-highlight {:on-press action-handler}
+                    [view st/toolbar-action
+                     [image action-image]]])
+             {:key (str "action-" action-image)}))
+         custom-action)]]
+     [sync-state-gradient-view]
+     [view st/toolbar-border-container
+      [view st/toolbar-border]]]))
+
+(defn- toolbar-search-submit [on-search-submit]
+  (let [text @(subscribe [:get-in [:toolbar-search :text]])]
```

Comment:
Is there a way to not use subscription here? It doesn't look like the best solution